framework/include/multiapps/MultiApp.h

```diff
@@ -96,12 +96,21 @@ class MultiApp : public MooseObject, public SetupInterface, public Restartable
   virtual bool solveStep(Real dt, Real target_time, bool auto_advance = true) = 0;
 
   /**
-   * Actually advances time and causes output.
-   *
-   * If auto_advance=true was used in solveStep() then this function
-   * will do nothing.
+   * Advances the multi-apps time step which is important for dt selection.
+   * (Note this does not advance the *time*. That is done in Transient::endStep,
+   * which is called either directly from solveStep() for loose coupling cases
+   * or through finishStep() for Picard coupling cases)
+   */
+  virtual void incrementTStep() = 0;
+
+  /**
+   * Calls multi-apps executioners' endStep and postStep methods which creates output and advances
+   * time (not the time step; see incrementTStep()) among other things. This method is only called
+   * for Picard calculations and is called *after* the master application's executioner's endStep
+   * method. Consequently Picard may not work in conjunction with restart since the sub-app's time
```

Comment:
Hmm, Are you sure about this statement? We do have a few Picard tests in the suite and this PR just passed them with recover.  Note that the master writes all of the recovery data for the simulation so as long as things are in the same state they were written in, things generally work out. If you still expect it to fail, we need to dive in and look at this more carefully.

Reply:
Yep, turns out Picard does work, although for a bit of a convoluted reason...The time of the sub-app *is* written out behind the time of the master, e.g. for a 20 step simulation with `dt = 0.1` and running `--half-transient`, the restartable data has the master's time as 1, and the stored sub-apps' time as 0.9. However, upon recover, the sub-app's time is caught-up in `Transient::incrementStepOrReject` because `_picard_max_its > 1` before any solves are performed.